src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -73,6 +73,50 @@
 
     }
 
+    private static final Map<String, String> TIME_ZONE_DOUBLE_TO_ID = new HashMap<String, String>();
+
+    static {
+        mapDoubleToId("-12.0", "Etc/GMT+12");
+        mapDoubleToId("-11.0", "US/Samoa");
+        mapDoubleToId("-10.0", "US/Hawaii");
+        mapDoubleToId("-9.5", "Pacific/Marquesas");
+        mapDoubleToId("-9.0", "US/Alaska");
+        mapDoubleToId("-8.0", "America/Los_Angeles");
+        mapDoubleToId("-7.0", "America/Phoenix");
+        mapDoubleToId("-6.0", "America/Chicago");
+        mapDoubleToId("-5.0", "America/New_York");
+        mapDoubleToId("-4.5", "America/Caracas");
+        mapDoubleToId("-4.0", "America/Halifax");
+        mapDoubleToId("-3.5", "America/St_Johns");
+        mapDoubleToId("-3.0", "America/Sao_Paulo");
+        mapDoubleToId("-2.0", "America/Noronha");
+        mapDoubleToId("-1.0", "Atlantic/Cape_Verde");
+        mapDoubleToId("0.0", "UTC");
+        mapDoubleToId("1.0", "Europe/Paris");
+        mapDoubleToId("2.0", "Europe/Athens");
+        mapDoubleToId("3.0", "Africa/Nairobi");
+        mapDoubleToId("3.5", "Asia/Tehran");
+        mapDoubleToId("4.0", "Asia/Dubai");
+        mapDoubleToId("4.5", "Asia/Kabul");
+        mapDoubleToId("5.0", "Asia/Tashkent");
+        mapDoubleToId("5.5", "Asia/Colombo");
+        mapDoubleToId("5.75", "Asia/Kathmandu");
+        mapDoubleToId("6.0", "Asia/Almaty");
+        mapDoubleToId("6.5", "Asia/Rangoon");
+        mapDoubleToId("7.0", "Asia/Jakarta");
+        mapDoubleToId("8.0", "Asia/Singapore");
+        mapDoubleToId("8.75", "Australia/Eucla");
+        mapDoubleToId("9.0", "Asia/Tokyo");
+        mapDoubleToId("9.5", "Australia/Adelaide");
+        mapDoubleToId("10.0", "Australia/Canberra");
+        mapDoubleToId("10.5", "Australia/Lord_Howe");
+        mapDoubleToId("11.0", "Pacific/Noumea");
+        mapDoubleToId("12.0", "Pacific/Auckland");
+        mapDoubleToId("12.75", "Pacific/Chatham");
+        mapDoubleToId("13.0", "Pacific/Tongatapu");
+        mapDoubleToId("14.0", "Pacific/Kiritimati");
```

Comment:
This may not be the best way to translate the double to a TimeZone object. I tested with some of the time zones which corresponded to the US cities in the list and DST was taken in account (eg. UTC -0800 shows up as -0700), which may be unexpected for users. 

https://docs.oracle.com/javase/7/docs/api/java/util/TimeZone.html 
Using the custom time zone ID may be more appropriate here, and it can be generated by code as well.